test/services/import_labels/test_import.py

```diff
@@ -0,0 +1,209 @@
+"""
+Tests the label import service
+"""
+import csv
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from test.services.import_labels.test_cases import TEST_CASES
+from typing import Any, Dict, List
+from unittest.mock import MagicMock
+
+import pytest
+
+from kili.services import import_labels_from_files
+from kili.services.label_import.importer import YoloLabelImporter
+from kili.services.label_import.parser import YoloLabelParser
+from kili.services.label_import.types import Classes
+from kili.services.types import ProjectId
+
+
+def _generate_label_file_list(
+    labels_data: List[Dict[str, Any]], filename: str, headers: List[str], annotation_root: str
+):
+    with Path(filename).open("w", encoding="utf-8") as l_d:
+        wrt = csv.writer(l_d)
+        wrt.writerow(headers)
+        for label_data in labels_data:
+            row = []
+            for h in headers:
+                row.append(
+                    label_data[h]
+                    if h != "path"
+                    else str(Path(annotation_root).joinpath(label_data[h]))
+                )
+            wrt.writerow(row)
+
+
+def _generate_label_file(yolo_rows: List[List], filename: str):
+    with Path(filename).open("w", encoding="utf-8") as y_f:
+        wrt = csv.writer(y_f, delimiter=" ")
+        wrt.writerows((str(a) for a in r) for r in yolo_rows)
+
+
+@pytest.mark.parametrize(
+    "description,inputs,outputs",
+    [
+        (test_case["description"], test_case["inputs"], test_case["outputs"])
+        for test_case in TEST_CASES
```

Comment:
remark for later: I think that from now, we should stop working with TEST_CASES list and rather use separete unittest TestCase classes. Otherwise it is very hard to know which one is failing and it is hard to understand the tests

Reply:
@parametrize creates one unit test per test case

Reply:
but I agree it can be discussed